Makefile

```diff
@@ -0,0 +1,76 @@
+# Copyright (C) Greenbone Networks GmbH
+# SPDX-License-Identifier: AGPL-3.0-or-later
+#
+# Build and development command for the scan management backend project.
+# important commands are documented and machine-runnable at the same time here.
+
+PROJECT := go-project-template
+GOPATH := $(shell go env GOPATH)
+
+.PHONY: help
+# Show this help
+help: # So I managed to get the help printing with the commment line above it
+	@awk '/^#/{c=substr($$0,3);next}c&&/^[[:alpha:]][[:alnum:]_-]+:/{print "\033[36m", substr($$1,1,index($$1,":")), "\033[0m", c}1{c=0}' $(MAKEFILE_LIST) | column -s: -t
+
+.PHONY: project-setup
+# Setup project for development. Install project dependencies, etc.
+project-setup: debian-os-package-install python-package-install go-package-install
+
+.PHONY: debian-os-package-install
+# Install debian apt packages (requires sudo)
+debian-os-package-install:
```

Comment:
one thing I did not implement myself yet in our Makefiles but am not super happy with is that thgis fails on non-debian/non-Linux OS... 
if you have time / an idea to solve it, it would be nice to make this fail with a clear message when users with a mac or non-apt-linux try to run it... one could even offer some automatic way to support Mac if this is a platform many use in the company by automatically checking this and setting variables/calling the proper target by that... just an idea, just if there is enough need...

Reply:
The more I think about it, the less I think this target is useful at all.
There are too many OSs and ways to install go. Also when developing a golang project, you probably should already set up your go environment beforehand ...